src/EbmlString.cpp

```diff
@@ -109,24 +109,20 @@ filepos_t EbmlString::ReadData(IOCallback & input, ScopeMode ReadFully)
     return GetSize();
 
   if (GetSize() == 0) {
-    Value = "";
-    SetValueIsSet();
+    Value.clear();
+
   } else {
-    auto Buffer = (GetSize() + 1 < std::numeric_limits<std::size_t>::max()) ? new (std::nothrow) char[GetSize() + 1] : nullptr;
-    if (Buffer == nullptr) {
-      // unable to store the data, skip it
-      input.setFilePointer(GetSize(), seek_current);
-    } else {
-      input.readFully(Buffer, GetSize());
-      if (Buffer[GetSize()-1] != '\0') {
-        Buffer[GetSize()] = '\0';
-      }
-      Value = Buffer;
-      delete [] Buffer;
-      SetValueIsSet();
-    }
+    Value.resize(GetSize());
+    std::memset(&Value[0], 0, GetSize());
```

Comment:
As the name suggests, readFully will read GetSize() bytes. So this memset is useless.
But not only that, resize actually fills the 0 for you: https://en.cppreference.com/w/cpp/string/basic_string/resize

We may use [reserve()](https://en.cppreference.com/w/cpp/string/basic_string/reserve) to avoid this double filling.

Reply:
`readFully` might reach the end of the file before it could read everything. I'm trying to avoid having remnants from the previous value still present in the case of failure.

`resize` only fills those parts with 0 that were newly added, meaning if new size > old size the bytes above old size will be set, nothing else. If `readFully` can only read some bytes < old size, there'll still be content from the previous value left.

This here is analog to what I did in `EbmlUnicodeString`, the difference being that there I have to convert to `std::wstring` anyway, meaning I cannot read into the final storage.

Reply:
If it reaches the end of file without filling the whole buffer, it will throw a runtime_error, which is not handled here. So if the function ends normally, the buffer *is* fully filled with the size requested.

Anyway, we can do proper optimizations later.